File: _includes/JS/solver.js
/**
* Reverse string
*
*  Args:
*   to_reverse: String to be reversed.
*
*  Returns:
*   Reversed string, in the format of a json object.
*/
function solve_reverse_string(to_reverse) {
  return str.split('').reverse().join('');
}

/**
* Find index of needle in Haystack array.
*
*  Args:
*   needle: String to be found.
*   haystack: Array with values to be searched.
*
*  Returns:
*   Index of the position of the needle.
*/
function solve_haystack(needle, haystack) {
  return haystack.indexOf(needle);
}

/**
* Find all the words in list that don't start with given prefix.
*
*  Args:
*   prefix: Prefix to be found.
*   strings: Array with values to be searched.
*
*  Returns:
*   List filtered with only values that `don't` start with prefix.
*/
function solve_prefix(prefix, strings) {
    return strings.filter(
        function(element) {
            if(element.substring(0, prefix.length) != prefix)
                return element;
        });
}

/**
* Return date increased with interval in the ISO 8601 format.
*
*  Args:
*   needle: Interval to be added to time.
*   datestamp: Initial date.
*
*  Returns:
*   The new data in the ISO 8601 format.
*/
function solve_dating(interval, datestamp) {
  var newDate = new Date(datestamp);
  newDate
    .setSeconds(newDate.getSeconds() + interval);
  return newDate.toISOString();
}

File: CLI/solver.py
from datetime import datetime, timedelta

def solve_reverse_string(to_reverse):
    ans = {'string': to_reverse[::-1]}

    return ans

def solve_haystack(input_dictionary):
    needle = input_dictionary['needle']
    haystack = input_dictionary['haystack']

    # It's guranteed that there is a needle on the haystack
    index = haystack.index(needle)

    ans = {'needle': index}

    return ans

def solve_prefix(input_dictionary):
    prefix = input_dictionary['prefix']
    array = input_dictionary['array']

    array = [string for string in array if not string.startswith(prefix)]

    ans = {'array': array}

    return ans

def solve_dating(input_dictionary):
    datestamp = input_dictionary['datestamp']
    interval = input_dictionary['interval']

    # Convert data to datetime format
    datestamp = datetime.strptime(datestamp, '%Y-%m-%dT%H:%M:%S.%fZ')
    interval = timedelta(0, interval)

    # Calculate & format answer
    ans = datestamp + interval
    ans = ans.strftime('%Y-%m-%dT%H:%M:%S.%f')[:-3] + 'Z'
    ans = {'datestamp': ans}

    return ans
